hal/ia32/cpu.h

```diff
@@ -36,66 +39,84 @@ static inline void hal_cpuInstrBarrier(void)
 }
 
 
-static inline unsigned char hal_inb(void *addr)
+static inline u8 hal_inb(void *addr)
 {
-	unsigned char val;
+	u32 ioport = (u32)addr;
```

Comment:
IMHO double cast here is somewhat better (like you did in the kernel). Perhaps move API `void *` -> `u16`, if easily done